### PR TITLE
Replace bash scripts for preview and publish modules with native python code

### LINKED ISSUE #56: Replace bash script calls with native Python implementation in preview_module
## Problem

The `preview_module.py` command currently calls external bash scripts from the Facets schemas repository using `curl` and `bash`. This approach creates several issues:

1. **Cross-platform compatibility**: Requires bash on Windows (Git Bash dependency)
2. **External dependencies**: Relies on external scripts that could change or become unavailable
3. **Error handling**: Limited control over error handling and debugging when scripts fail
4. **Security**: Downloading and executing scripts from remote URLs
5. **Maintenance**: Harder to maintain and debug when issues occur

## Current Implementation

The code currently calls two bash scripts:

### 1. Module Registration/Preview
```bash
curl -s https://facets-cloud.github.io/facets-schemas/scripts/module_register.sh | bash -s -- [args]
```

### 2. Module Publishing  
```bash
curl -s https://facets-cloud.github.io/facets-schemas/scripts/module_publish.sh | bash -s -- [args]
```

### Move script logic to Python
1. **Replace `module_register.sh` logic** with native Python HTTP calls
2. **Replace `module_publish.sh` logic** with native Python HTTP calls  
3. **Remove bash/curl dependencies** for core functionality
4. **Implement proper error handling** with detailed Python exception handling
5. **Add comprehensive logging** for better debugging

### Benefits of Python Implementation
- **Cross-platform compatibility**: No more Git Bash requirement on Windows
- **Better error handling**: Detailed Python exceptions and error messages
- **Improved debugging**: Direct control over HTTP requests and responses
- **Reduced dependencies**: No external script dependencies
- **Enhanced security**: No remote script execution
- **Better testing**: Unit testable Python code vs external scripts

## Implementation Approach

1. **Analyze current bash scripts** to understand the HTTP API calls being made
2. **Implement equivalent Python functions** using `requests` library
3. **Maintain same CLI interface** and behavior
4. **Add comprehensive error handling** for HTTP failures
5. **Include detailed logging** for debugging
6. **Update tests** to cover new Python implementation

## Files to Update

- `ftf_cli/commands/preview_module.py` - Main implementation
- Add new utility functions for HTTP API calls
- Update any related tests
- Update documentation if needed

## Impact

This change will significantly improve the reliability and maintainability of the module preview/publish functionality, especially on Windows platforms.

### PR DESCRIPTION
**Pull request by anujhydrabadi**
Fixes #56 

## Summary by CodeRabbit

- **Refactor**
  - Streamlined the module preview command for improved reliability and maintainability by using internal Python functions instead of shell commands.
- **New Features**
  - Introduced new operations for registering and publishing Terraform modules directly with a control plane.
- **Bug Fixes**
  - Enhanced error handling and user feedback during module registration and publishing.
- **Tests**
  - Added comprehensive tests for module operations and the preview command, including edge cases and error scenarios.
- **Chores**
  - Updated ignore rules to exclude local settings files from version control.
- **Documentation**
  - Updated minimum required Python version in installation instructions and package configuration.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -26,7 +26,7 @@ To install FTF CLI from source, follow these steps:
 
 #### Prerequisites
 
-- Python 3.9 or later
+- Python 3.11 or later
 - Virtual environment (recommended)
 - [jq](https://github.com/jqlang/jq)
 - [Git Bash](https://winget.run/pkg/Git/Git)
```
Comment (sanmesh-kakade) [inline README.md:32]: you can remove this if not required
